pkg/postgres/error.go

```diff
@@ -11,6 +11,8 @@ func toErrox(err error) error {
 		switch pgErr.Code {
 		case "23505":
 			return errors.Wrap(errox.AlreadyExists, err.Error())
+		case "23503":
+			return errors.Wrap(errox.ReferencedByAnotherObject, err.Error())
```

Comment:
`23503` is `foreign_key_violation` and it is not only about deletion. It is also about insertion - an attempt to insert a record with a foreign key that doesn't match any existing key will fail. The current `ReferencedByAnotherObject` does not tell about that case.

Reply:
Added conditional matching here based on the pg error's detail - sadly no distinct error types are being returned.

Added a test to cover this as well.